### PR TITLE
Fixed creating built-in validator in model with same function name

### PR DESCRIPTION
**Pull request by dmirogin**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14370

### CODE REVIEW

File: tests/data/validators/models/FakedValidationModel.php [hunk 1]
```diff
@@ -88,4 +88,12 @@ public function getInlineValArgs()
     {
         return $this->inlineValArgs;
     }
+
+    /**
+     * @see https://github.com/yiisoft/yii2/issues/14370
+     */
+    private function number()
```
Comment (dynasource) [inline tests/data/validators/models/FakedValidationModel.php]: shouldnt this be mapped to a configured rule `Rules()`?

**Review by dynasource — CHANGES_REQUESTED** [review]

File: tests/data/validators/models/FakedValidationModel.php (hunk 1 above)
Comment (dmirogin) [inline tests/data/validators/models/FakedValidationModel.php]: This bug happen when you have function in model class with name that same to one of built-in validators. And Validator::createValidator create InlineValidator instead NumberValidator in this case. I can make separate model class and test for getting this more clear.

Comment (dynasource) [inline tests/data/validators/models/FakedValidationModel.php]: true. It's a bit magical in the current form

**Review by dynasource — APPROVED** [review]
looks more verbose now

**Review by cebe — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**